### PR TITLE
Refactor electron.profile and electron based programs

### PR DESCRIPTION
**Pull request by rusty-snake**
see  #3806

### CODE REVIEW

File: etc/profile-a-l/freetube.profile [hunk 1]
```diff
@@ -6,26 +6,22 @@ include freetube.local
 # Persistent global definitions
 include globals.local
 
+# Disabled until someone reported positive feedback
```
Comment (glitsj16) [inline etc/profile-a-l/freetube.profile:9]: I have been running freetube without issues having these in my freetube.local:

include whitelist-common.inc
include whitelist-runuser-common.inc
include whitelist-usr-share-common.inc
include whitelist-var-common.inc

nou2f
novideo

The nou2f is untested due to not having such hardware.

Comment (rusty-snake) [reply etc/profile-a-l/freetube.profile:9]: If I'm right freetube has no login support at all.

File: etc/profile-m-z/riot-web.profile [hunk 2]
```diff
@@ -4,14 +4,28 @@
 # Persistent local customizations
 include riot-web.local
 # Persistent global definitions
-# added by included profile
-#include globals.local
+include globals.local
+
+# Disabled until someone reported positive feedback
```
Comment (glitsj16) [inline etc/profile-m-z/riot-web.profile:9]: Tested with element-{desktop,web} from Arch community repo. Works fine with the below in riot-web.local:

```

# WIP 3807
include disable-devel.inc
# shell needed for tray icon
noblacklist ${PATH}/bash
noblacklist ${PATH}/sh
ignore noexec /tmp
whitelist /usr/share/chromium
whitelist /usr/share/webapps/element
include disable-exec.inc
include disable-interpreters.inc
include disable-xdg.inc
include whitelist-runuser-common.inc
include whitelist-usr-share-common.inc
include whitelist-var-common.inc
nou2f
novideo
# shell needed for tray icon
#shell none
disable-mnt
private-cache
private-dev
private-tmp
```

Comment (rusty-snake) [reply etc/profile-m-z/riot-web.profile:9]: two questions
- electron does not `include disable-shell.inc`, is it in one of your locals?
- `shell none` just mean start direct and w/o a shell, but a shell is still present inside the sandbox. Does it really need to be ignored?

Comment (glitsj16) [reply etc/profile-m-z/riot-web.profile:9]: > electron does not include disable-shell.inc, is it in one of your locals?

It isn't, nice catch. The relatively long include chain (element-desktop --> riot-desktop --> riot-web --> electron) threw me off here apparently.

> shell none just mean start direct and w/o a shell, but a shell is still present inside the sandbox. Does it really need to be ignored?

Ditto as above. I suspect it was a leftover from testing permutations. I can now confirm adding 'shell none' works fine.

To put things straight, element-desktop is working with the below in my riot-web.local:

# WIP 3807
include disable-devel.inc
ignore noexec /tmp
whitelist /usr/share/chromium
whitelist /usr/share/webapps/element
include disable-exec.inc
include disable-interpreters.inc
include disable-xdg.inc
include whitelist-runuser-common.inc
include whitelist-usr-share-common.inc
include whitelist-var-common.inc
nou2f
novideo
shell none
disable-mnt
private-cache
private-dev
private-tmp

**Comment by mYnDstrEAm** [comment]
Should it be sufficient to change /etc/firejail/electron.profile riot-web.profile and riot-desktop.profile to get element-desktop working with the latest firejail version in backports in Debian 10 (0.9.64)? I changed these 3 files as well as element-desktop.profile to match the latest versions in this repo but now I get this error when running `firejail element-desktop`:
```
Warning: An abstract unix socket for session D-BUS might still be available. Use --net or remove unix from --protocol set.
Child process initialized in 1135.90 ms
The setuid sandbox is not running as root. Common causes:
  * An unprivileged process using ptrace on it, like a debugger.
  * A parent process set prctl(PR_SET_NO_NEW_PRIVS, ...)
Failed to move to new namespace: PID namespaces supported, Network namespace supported, but failed: errno = Operation not permitted

Parent is shutting down, bye...
```

**Comment by rusty-snake** [comment]
Using profiles from git master with a stable release (even the latest) is nothing what you should expect to work. Anyway _ATM_ it works to use git master profiles in 0.9.64.

debian + element-desktop -> #3586.

**Pull request merged**